association_backend/association/webhooks/views.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from typing import cast
+
+import stripe
+from association.db import get_engine, get_session
+from association.domain import services
+from association.domain.exceptions import SubscriptionNotCreated, SubscriptionNotUpdated
+from association.domain.repositories import AssociationRepository
+from association.domain.services import SubscriptionInputModel, SubscriptionUpdateInput
+from association.settings import STRIPE_WEBHOOK_SIGNATURE_SECRET
+from sqlalchemy.ext.asyncio import AsyncSession
+from starlette.endpoints import HTTPEndpoint
+from starlette.responses import JSONResponse
+
+logger = logging.getLogger(__name__)
+
+
+class StripeWebhook(HTTPEndpoint):
+    """
+    https://stripe.com/docs/billing/subscriptions/checkout#webhooks
+    """
+
```

Comment:
I wonder if there are better way to write this code, have you considered using function based routes? it might just be enough to make the code look better

Reply:
It is very hard for me because I love Class Based Views